### ISSUE TITLE
Review projects with ChatGPT based on Contributing Guide

### ISSUE THREAD

**Issue by Ly0n** [protontypes/open-sustainable-technology#896]
The following prompt worked very well for 4 open source projects I have tested: 

> Could you please review the following open source project to see if it meets the Contribution Guide ( https://opensustain.tech/contributing/) of https://opensustain.tech: https://github.com/OceanParcels/plasticparcels

**ChatGPT**:

Searched 1 site

To meet the contribution criteria for Open Sustainable Technology, a project must be actively developed or used, show external contributions (like pull requests or issues from others), have clear documentation to support reuse, and be published under an open-source license. Additionally, it should align with the mission of preserving ecosystems through open technology. If a project is in early development, it can still be submitted with the label "Under Observation."

Let me now review the specific project "plasticparcels" to see if it fits these guidelines.

Searched 1 site

The "plasticparcels" repository on GitHub aligns well with the contribution guide of Open Sustainable Technology. It is an actively maintained open-source project with external contributions, such as pull requests and issues. It has clear documentation, examples, and tutorials for reuse, as well as an MIT license that ensures open access. The project focuses on simulating ocean plastic dispersion, fitting within the ecosystem preservation goals of OpenSustain.Tech.

Based on this, it meets the criteria for contributions to the platform.

**Comment by Ly0n** [comment]
I have thought a lot about whether we should allow ChatGPT to take this central position in the project. In my opinion, there is a big risk that people will not like this project in the near future because they will have the impression that the content is generated/curated by AI. I suspect that people will soon be able to create similar lists with just a single question. The only way to stand out from the crowd in the future is to stand out from the large amount of AI-generated content that is soon to be expected.

 So integrating a bot to help us review might not be the best idea. The benefit is very small.

I see an important application in the automatic discovery of new candidates that can then be reviewed by a human. This will just open up another source of projects besides manual investigation.

**Comment by VidhyaDivakar** [comment]
Hi @Ly0n, does this require further review or research. If yes, can I work on this? If no, can you suggest me any other doc issues that I can contribute on. I am a Senior Technical Writer with 8 plus years of experience in software documentation and publishing.

Thanks,
Vidhya

**Comment by Ly0n** [comment]
Yes, that would be fantastic and a great help to us, @VidhyaDivakar. We welcome any method that helps us find more projects. I'm sure you'll find other ways to find projects using ChatGPT and other LLMs. This will also help you to find projects that you can contribute to the dataset. Please also take a look at how we currently recommend investigating new projects using LLMs. I bet much more can be automated here: https://github.com/protontypes/open-sustainable-technology/blob/main/docs/how_to_identify_projects.md

Please let me know if you need any support with the review, or if you would like to discuss some ideas. You are also welcome to join the our community chat here: https://discord.gg/BTYjRSvUBB

**Comment by VidhyaDivakar** [comment]
Thanks @Ly0n. For now, I am starting to look at #896. I will reach out to you if I need help. Please change the assignee's name to my name.

Yes, thanks for the invite; I added myself to the community chat Discord, it's really great!

Thanks,
Vidhya

**Comment by VidhyaDivakar** [comment]
@Ly0n, I have come up with a template.md for Reviewing Project and based on that template, reviewed prasticparcels and created a report.md file. 

The template, which I have come up with can be used to review projects going forward. 

I have not used LLMs for this, I have come up with the review report manually.

Can you tell me in which folder to place these two files for your review? (Right now they are in my local). 

Thanks,
Vidhya

**Comment by VidhyaDivakar** [comment]
@Ly0n, can you take a look at this? Where shall I put the report and the report template?

Just adding the whole report here.

# OpenSource Sustainability check on the Project Plastic Parcel

**Summary**
Plastic Parcel claims that the parcels can be used to track both passive and active particulates such as water, plankton, plastics, icebergs, and fish—using data from flow fields. Parcels provide a set of Python classes and methods to create customizable particle tracking simulations. In this report you will see whether this project falls under the category of Open Sustainability.

## **The Journal of Open Source Software checklist**

Are there any journal related to this project in the Journal of Open Source Software Checklist.

Yes this project is listed in [Journal of Open Source](https://joss.theoj.org/papers/10.21105/joss.07094/) in the year 2024.

- License - Authors of JOSS papers retain copyright.
- This work is licensed under a Creative Commons Attribution 4.0 International License.

**Other notable listing**
This is based on a paper that was submitting to [European Geoscience union](https://gmd.copernicus.org/articles/10/4175/2017/)  in 2017 and 2019. The paper descriptions covers two different  versions v.09 and v2.0

## **Allignment with the [Open Sustainability Principle](https://opensustain.tech/blog/open_principles_for_a_sustainable_technology_transition/?h=open+principles+sustainability)**

1. #### **Transparency and Trust**

   Open practices can be crucial to ensure transparency and trust in finding answers; based on this point, the following are the thinfs to note:


   - This is project is transparent as it has the [source code](https://github.com/Parcels-code/plasticparcels/releases), APIs accessibtle in the open source platform. Its open source code is available for pull request and forking.
   - However, this project does not have a security.md file yet.

2. #### **Knowledge Evolution and Adaptation**

   Opening knowledge and technology will make our economy more sustainable, based on this, the following points make this project sustainable:

   - The plasticparcels can be installed from [conda-forge](https://anaconda.org/channels/conda-forge/packages/plasticparcels/overview), a community-led collection of recipes, build infrastructure, and distributions for the conda package manager.
     - **License** : [MIT](https://github.com/Parcels-code/plasticparcels?tab=MIT-1-ov-file#readme)
     - Total Downloads: 4.3K
     - Platforms - noarch Version: 0.3.1
       A noarch package is a package that doesn’t depend on the system’s CPU architecture
   - Involves structured documentation for easy accessibility and extendability.

3. #### Colloborative Innovation

   Open innovation strategy evolves by colloborative approach, based on this, the following are the things to consider about PlasticParcels:
   - This project has one contributor outside of the org level.
   - Has a few open issues, for the contributors to work on.
   - Has four pull requests done
   - Also, has few open forum discussions about the project in the open source platform
4. #### Open Business Models

   The important aspect of any sustainable project is its open business modes. Being an intro-level project, plasticparcels do have open business models with certain gaps.

#### **The Scope of the Project**

- Does the project support preserving the natural ecosystem?
Yes, this project talks about
- Is it actively developed by users outside the core project or organization
- Involves structured documentation for easy accessibility and extendability.

  - Read.me - Yes, Read Me file is present (documented gaps in the Gaps in the Project section)
  - Contributing.md - No, this is not present as a separate file. This is just given as a section in the ReadMe file.
  - Folder Structure - project-name/project-name/
    ├── .github/
    │   ├── actions/
    │   │   └── install-plasticparcels/
    │   │       └── action.yml
    │   └── workflows/
    │       ├── draft-pdf.yml
    │       └── unit_tests.yml
    │
    ├── .docs/
    │   ├──_static/
    │   │   ├── plasticparcelslogo.png
    │   │   └── plasticparcelslogo_inverted.png
    │   │
    │   ├── examples/
    │   │   └── ci.yml
    │   │
    │   ├── paper/
    │   │
    │   └── reference/
    │       ├── plasticparcelslogo.png
    │       └── plasticparcelslogo_inverted.png
    │
    ├── .git_archival.txt
    ├── .gitattributes
    └── .git/   (and other git internal files...)
  - Automation Scripts, scope - Yes present
  - Has [MIT&#39;s open source license](https://github.com/Parcels-code/plasticparcels?tab=MIT-1-ov-file#readme)
- What is the security criteria provided? - This is not clear so far.

#### **Gaps in the Project**

1. There are a few broken links in the ReadMe file under the Required Data section:

- In the line, "plasticparcels has been developed for use with data from the Copernicus Marine Service, and requires the following data to run:"

  - Hydrodynamic model data: MOI GLO12 (psy4v3r1)
  - Biogeochemical model data: MOI BIO4 (biomer4v2r1)

    ***These two lines return 404 error.***

2. The Contributing file is not present. It is just given as a section in the ReadMe.md file.

#### Sustainability keywords

Is the project consisting of the keywords listed in the [ost_keywords.txt]([https://raw.githubusercontent.com/protontypes/osta/refs/heads/main/ost_keywords.txt](https://raw.githubusercontent.com/protontypes/osta/refs/heads/main/ost_keywords.txt)).

Keywords check: Here are the few words that I checked in the repo of the plastic parcels, and here are the results in Boolean:

asymmetric – yes, biodiversity – no, climate – yes, cloud – yes, earth – yes, emission – yes, energy – no, evolution – no, farm – no, green house – no, hazard – no, human – no, impact – yes, metric – no, natural – yes, ocean – yes, plastic – yes, publicly – no, satellite – no, science – yes, science – yes, soil – no, solar – no, sustainable – no, tree – yes, validation – no, water – yes, weather – no, wind – yes

##### Open Source Keywords

Is the project consisting of the keywords such as "open-source software," "open data," "GitHub repository,"  "Open-source-tools", "contributing," etc - Yes.

##### **Package Level Research**

What do the libraries, methods, and objects of the projects talk about? Are they related to sustainability?

There is not much package information in this project.

Thanks,
Vidhya
